components/nexus-security/src/main/java/org/sonatype/security/DefaultSecuritySystem.java

```diff
@@ -664,101 +618,11 @@ public void changePassword(String userId, String newPassword)
     eventBus.post(new UserPrincipalsExpired(userId, user.getSource()));
   }
 
-  public void forgotPassword(String userId, String email)
```

Comment:
This feature was already removed from the UI (and thus the product), this was only removing the underpinnings.

https://issues.sonatype.org/browse/NEXUS-7294 is to replace this with something sane and secure.